sound/soc/intel/boards/sof_maxim_common.c

```diff
@@ -326,17 +381,28 @@ const struct snd_soc_ops max_98390_ops = {
 };
 EXPORT_SYMBOL_NS(max_98390_ops, SND_SOC_INTEL_SOF_MAXIM_COMMON);
 
-void max_98390_set_codec_conf(struct snd_soc_card *card, int ch)
+void max_98390_dai_link(struct snd_soc_dai_link *link)
+{
+	link->num_codecs = get_maxim_num_codec();
+	if (link->num_codecs == 4)
+		link->codecs = max_98390_4spk_components;
+
+	link->codecs = max_98390_components;
```

Comment:
Won't this overwrite `link->codecs = max_98390_4spk_components;` if link->num_codecs == 4?